### PR TITLE
drivers: adc: adc_ads1119: Fix configuration register access.

### LINKED ISSUE #70091: ADS1119 configuration register access macros do not work
**Describe the bug**
Following macros in drivers/adc/ads1119.c always produce result = 0, because bitmask is shifted first.
```
#define ADS1119_CONFIG_CM(x) ((x) & BIT(1))
#define ADS1119_CONFIG_DR(x) ((x) & (BIT_MASK(2) << 2))
#define ADS1119_CONFIG_GAIN(x) ((x) & BIT(4))
#define ADS1119_CONFIG_MUX(x) ((x) & (BIT_MASK(3) << 5))
```

Example:
```
enum {
	ADS1119_CONFIG_DR_20            = 0,
	ADS1119_CONFIG_DR_90            = 1,
	ADS1119_CONFIG_DR_330           = 2,
	ADS1119_CONFIG_DR_1000          = 3,
	ADS1119_CONFIG_DR_DEFAULT       = ADS1119_CONFIG_DR_20,
};
```
Calculation:
`ADS1119_CONFIG_DR(ADS1119_CONFIG_DR_90) -> (0x01) & (0x03 << 2) -> 0x01 & 0x0C -> 0x00`

**To Reproduce**
1. Try to change ADS1119 configuration register.

**Expected behavior**
With following change the macros produce expected result. See adc_ads1112.c driver, which macros look like this as well.
```
#define ADS1119_CONFIG_CM(x) (((x) & BIT(0)) << 1)
#define ADS1119_CONFIG_DR(x) (((x) & BIT_MASK(2)) << 2)
#define ADS1119_CONFIG_GAIN(x) ((x) & BIT(0) << 4)
#define ADS1119_CONFIG_MUX(x) (((x) & BIT_MASK(3)) << 5)
```
Calculation:
`ADS1119_CONFIG_DR(ADS1119_CONFIG_DR_90) -> (0x01 & 0x03) << 2) -> 0x01 << 2 -> 0x04`

**Impact**
Cannot properly set ADS1119 configuration register. ADC always works with its default settings.

**Logs and console output**
No error logged.

**Environment (please complete the following information):**
* OS: Ubuntu 22.04.4 LTS
* Toolchain: Zephyr SDK v0.16.5
* Commit SHA or Version used: v3.6.0

**Additional context**
Nothing specific.

### PR DESCRIPTION
**Pull request by jan-kubiznak**
The adc_ads1119 driver is unable to overwrite the configuration register, the chip therefore always works with its default settings. Register access macros are fixed by this commit.

Fixes: #70091

### CODE REVIEW

**Review by aasinclair — DISMISSED** [review]

**Review by aasinclair — DISMISSED** [review]

File: drivers/adc/adc_ads1119.c [hunk 1]
```diff
@@ -22,10 +22,10 @@ LOG_MODULE_REGISTER(ADS1119, CONFIG_ADC_LOG_LEVEL);
...
 
 #define ADS1119_CONFIG_VREF(x) ((x) & BIT(0))
-#define ADS1119_CONFIG_CM(x) ((x) & BIT(1))
-#define ADS1119_CONFIG_DR(x) ((x) & (BIT_MASK(2) << 2))
-#define ADS1119_CONFIG_GAIN(x) ((x) & BIT(4))
-#define ADS1119_CONFIG_MUX(x) ((x) & (BIT_MASK(3) << 5))
+#define ADS1119_CONFIG_CM(x) (((x) & BIT(0)) << 1)
+#define ADS1119_CONFIG_DR(x) (((x) & BIT_MASK(2)) << 2)
+#define ADS1119_CONFIG_GAIN(x) (((x) & BIT(0)) << 4)
+#define ADS1119_CONFIG_MUX(x) (((x) & BIT_MASK(3)) << 5)
```
Comment (decsny) [inline drivers/adc/adc_ads1119.c:28]: perhaps use FIELD_PREP? https://docs.zephyrproject.org/latest/doxygen/html/group__sys-util.html#gaa03c8b31bf67a097dd9f8153a04ef86b

Comment (jan-kubiznak) [reply drivers/adc/adc_ads1119.c]: Changed in [118ab41](https://github.com/zephyrproject-rtos/zephyr/pull/70120/commits/118ab41b80b995bf1c6fe3805aa683617b20a636).

File: drivers/adc/adc_ads1119.c [hunk 2]
```diff
@@ -21,11 +21,11 @@
...
 
-#define ADS1119_CONFIG_VREF(x) ((x) & BIT(0))
-#define ADS1119_CONFIG_CM(x) ((x) & BIT(1))
-#define ADS1119_CONFIG_DR(x) ((x) & (BIT_MASK(2) << 2))
-#define ADS1119_CONFIG_GAIN(x) ((x) & BIT(4))
-#define ADS1119_CONFIG_MUX(x) ((x) & (BIT_MASK(3) << 5))
+#define ADS1119_CONFIG_VREF(x) (FIELD_PREP(GENMASK(0, 0), x))
+#define ADS1119_CONFIG_CM(x) (FIELD_PREP(GENMASK(1, 1), x))
+#define ADS1119_CONFIG_DR(x) (FIELD_PREP(GENMASK(3, 2), x))
+#define ADS1119_CONFIG_GAIN(x) (FIELD_PREP(GENMASK(4, 4), x))
```
Comment (decsny) [inline drivers/adc/adc_ads1119.c:27]: sorry to be unclear, didn't mean to convert these BIT to GENMASK, the BIT and BIT_MASK made more sense, but keep the FIELD_PREP

Comment (jan-kubiznak) [reply drivers/adc/adc_ads1119.c]: Changed in [e0acd25](https://github.com/zephyrproject-rtos/zephyr/pull/70120/commits/e0acd2527e715df7dd20c9de4e772a64a21a6b0f).

**Review by decsny — APPROVED** [review]

**Review by aasinclair — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Pull request merged**